File: app/src/main/java/pl/coddev/applu/ui/UninstallWidgetS.java
package pl.coddev.applu.ui;

import android.content.Context;
import android.widget.RemoteViews;

import pl.coddev.applu.R;

/**
 * Created by Piotr Woszczek on 23/07/15.
 */
abstract public class UninstallWidgetS extends UninstallWidget {
    private static final String TAG = "UninstallWidgetS";

    void setupLastAppsButtons(int widgetId, RemoteViews views, Context context, int[] ids) {
    }

    ;

    void handleLastApps(String newPackage, int widgetId) {
    }

    ;

    //    void setupRemoveAllButton(RemoteViews views, Context context, int[] ids){};
    @Override
    void setupRemoveAllButton(RemoteViews views, Context context, int[] ids) {
        views.setOnClickPendingIntent(R.id.clearAllButton, buildPendingIntent(context, WidgetActions.BUTTON_CLEAR_ALL.name(), ids));
    }

    abstract void switchSelectorStatus(RemoteViews views);

    abstract RemoteViews getRemoteViews(Context context);


}

File: app/src/main/java/pl/coddev/applu/ui/UninstallWidgetS.kt
package pl.coddev.applu.ui

import android.content.Context
import android.widget.RemoteViews
import pl.coddev.applu.R
import pl.coddev.applu.enums.WidgetActions

/**
 * Created by Piotr Woszczek on 23/07/15.
 */
abstract class UninstallWidgetS : UninstallWidget() {
    override fun setupLastAppsButtons(widgetId: Int, views: RemoteViews, context: Context, ids: IntArray) {}
    fun addToLastApps(newPackage: String, widgetId: Int) {}

    //    void setupRemoveAllButton(RemoteViews views, Context context, int[] ids){};
    override fun setupRemoveAllButton(views: RemoteViews, context: Context, ids: IntArray) {
        views!!.setOnClickPendingIntent(R.id.clearAllButton, buildPendingIntent(context, WidgetActions.BUTTON_CLEAR_ALL.name, ids))
    }

    abstract override fun getRemoteViews(context: Context): RemoteViews

    companion object {
        private const val TAG = "UninstallWidgetS"
    }
}